crypto/x509/x509_req.c

```diff
@@ -286,6 +286,18 @@ void X509_REQ_get0_signature(const X509_REQ *req, const ASN1_BIT_STRING **psig,
         *palg = &req->sig_alg;
 }
 
+void X509_REQ_set0_signature(X509_REQ *req, ASN1_BIT_STRING *psig)
+{
+    if (req->signature)
+           ASN1_BIT_STRING_free(req->signature);
+    req->signature = psig;
+}
+
+int X509_REQ_set1_signature(X509_REQ *req, X509_ALGOR *palg)
```

Comment:
The @mattcaswell 's comment is still relevant. The name of the function is wrong. It should be X509_REQ_set1_sigalg or X509_REQ_set1_signature_algo or something similar.

Reply:
Missed that. Done & tested.